### PR TITLE
moving postgres to run on /mnt

### PR DESCRIPTION
**Pull request by JoeNunnelley**
@spaceballone first pass at this. left the /etc/postgresql/ config folder where it is since its just where the server config get read from during startup.

### CODE REVIEW

File: templates/default/postgresql.conf.erb [hunk 1]
```diff
@@ -0,0 +1,591 @@
+# -----------------------------
+# PostgreSQL configuration file
```
Comment (spaceballone) [inline templates/default/postgresql.conf.erb]: Wondering though if we really want to check this file in. After all, if we upgrade postgres we now have to come and remember to change the config file as well. Should we do something like a "sed" on the file that postgres lays down to just change this line?

File: recipes/_postgres.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+data_directory = '/var/lib/postgresql'
+override_data_directory = node['postgresql']['override_data_directory']
+version = node['postgresql']['version']
+
+directory override_data_directory 
+
+include_recipe 'postgresql'
+
+##reconfigure location
+bash 'move data folder' do
```
Comment (michaelb990) [inline recipes/_postgres.rb]: there should be no data here (these recipes are exclusively for the jenkins slaves), so is this part really necessary?

File: recipes/_postgres.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  code "sudo rsync -av #{data_directory} #{override_data_directory}"
+  only_if { ::File.exist?(data_directory) }
+end
+
+bash 'backup default install' do
```
Comment (michaelb990) [inline recipes/_postgres.rb]: again, this is for a packer build job on a fresh AMI, do we really need to be this careful?

File: recipes/_postgres.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+# update the postgresql.conf file to have the updated data_directory
+template "#{override_data_directory}/postgresql/#{version}/main/postgresql.conf" do
+  source 'postgresql.conf.erb'
+  owner 'postgres'
+  group 'postgres'
+end
+
+# update the postgresql.conf file to have the updated data_directory
+template "/etc/postgresql/#{version}/main/postgresql.conf" do
```
Comment (michaelb990) [inline recipes/_postgres.rb]: why do we do this 2x?

File: templates/default/postgresql.conf.erb (hunk 1 above)
Comment (JoeNunnelley) [inline templates/default/postgresql.conf.erb]: generalized this so moving from version to version should not be an issue.  I believe the accepted practise is to use template files and have the attributes we may change provided by  variables in the attributes files.

File: recipes/_postgres.rb (hunk 2 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: now it is removed.

File: recipes/_postgres.rb (hunk 3 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: nope, removed.

File: recipes/_postgres.rb (hunk 4 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: removed, the second change was not necessary.

**Comment by michaelb990** [comment]
What are you trying to accomplish with this? I'm not sure we need to do this for the jenkins workers running pg 9.4.

**Comment by michaelb990** [comment]
I'm also pretty sure that `/mnt` is the ephemeral instance store that gets attached to the instance on startup. It won't persist when we save the AMI and re-launch the instance as a jenkins worker.

**Comment by JoeNunnelley** [comment]
@michaelb990 I was requested to make the installation point for postgres a folder where an SSD drive could back it (turns out this is mapped to /data on ec2 instances with SSD drives that we use).  We are making this change with the hope that the jenkins workers will build and test core builds faster when the postgres instance is on an SSD.

**Comment by JoeNunnelley** [comment]
removed the special logic for installing in a non-standard location. that will do us no good if we use ephemeral drives in a secondary position (/data). We have to use a root SSD in which case the special installation point is actually not needed.

**Comment by JoeNunnelley** [comment]
@michaelb990 s'okay?

File: .kitchen.yml [hunk 5]
```diff
@@ -19,9 +19,3 @@ suites:
   - name: worker
     run_list:
       - jenkins-workers::worker
-
-  - name: elixir
```
Comment (michaelb990) [inline .kitchen.yml]: why is this being deleted?

File: recipes/_postgres.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+include_recipe 'jenkins-workers::_common'
+
+version = node['postgresql']['version']
+
+include_recipe 'postgresql'
```
Comment (michaelb990) [inline recipes/_postgres.rb]: why are we making things more complicated? before we just installed postgres with `package 'postgresql-N.N'`. also, do we still install postgis somewhere?

File: recipes/_postgres.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+# update the postgresql.conf file to have the updated data_directory
+template "/etc/postgresql/#{version}/main/postgresql.conf" do
+  source 'postgresql.conf.erb'
+  owner 'postgres'
+  group 'postgres'
+end
+
+bash 'start postgres' do
```
Comment (michaelb990) [inline recipes/_postgres.rb]: why?

Comment (michaelb990) [inline recipes/_postgres.rb]: I don't think we need to do this, but even if we did, 
```
service 'postgresql' do
  action :restart
end
``` 
or something similar seems like a better option.

File: recipes/_postgres.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+include_recipe 'jenkins-workers::_common'
+
+version = node['postgresql']['version']
+
+include_recipe 'postgresql'
+
+# update the postgresql.conf file to have the updated data_directory
+template "/etc/postgresql/#{version}/main/postgresql.conf" do
```
Comment (michaelb990) [inline recipes/_postgres.rb]: if we are going to use the postgres recipe, please just configure the data directory via the chef attribute

https://supermarket.chef.io/cookbooks/postgres

File: templates/default/postgresql.conf.erb [hunk 9]
```diff
@@ -0,0 +1,586 @@
+# -----------------------------
```
Comment (michaelb990) [inline templates/default/postgresql.conf.erb]: If you use the postgres recipe (this is a good argument to actually using it), you should configure the data directory with a chef attribute, not a copied / templated config file.

File: .kitchen.yml (hunk 5 above)
Comment (JoeNunnelley) [inline .kitchen.yml]: its a sub-recipe that is part of the worker. I didn't feel it warranted its own line anymore.

File: recipes/_postgres.rb (hunk 6 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: yes, I'm installing postgis as part of the client packages list (attributes/default.rb).  Part of this effort was moving to using publicly maintained cookbooks like 'postgresql' to try to minimize the amount of homegrown solutions we must maintain.

File: recipes/_postgres.rb (hunk 7 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: that is a better solution. this was a carry over from the initial recipe.  thanks!

File: recipes/_postgres.rb (hunk 8 above)
Comment (JoeNunnelley) [inline recipes/_postgres.rb]: done

File: templates/default/postgresql.conf.erb (hunk 9 above)
Comment (JoeNunnelley) [inline templates/default/postgresql.conf.erb]: done

File: .kitchen.yml (hunk 5 above)
Comment (michaelb990) [inline .kitchen.yml]: it's nice to have test suites around the partial recipes, so if you change just something in that recipe, you can iterate a bit faster.

File: attributes/default.rb [hunk 10]
```diff
@@ -49,6 +49,7 @@
 
 # Postgres Recipe
 default['postgresql']['version'] = '9.3'
+default['postgreqql']['dir'] = "/var/lib/postgresql/#{default['postgresql']['version']}/main"
```
Comment (michaelb990) [inline attributes/default.rb]: typo

File: attributes/default.rb [hunk 11]
```diff
@@ -46,3 +46,10 @@
 default['iss']['db']['name'] = 'iss_test'
 default['iss']['db']['host'] = 'localhost'
 default['iss']['db']['port'] = 5432
+
+# Postgres Recipe
+default['postgresql']['version'] = '9.3'
+default['postgreqql']['dir'] = "/var/lib/postgresql/#{default['postgresql']['version']}/main"
```
Comment (michaelb990) [inline attributes/default.rb]: isn't this already the default?

Comment (JoeNunnelley) [inline attributes/default.rb]: yes. it is.

File: recipes/_postgres.rb (hunk 6 above)
Comment (michaelb990) [inline recipes/_postgres.rb]: 👍

File: attributes/default.rb (hunk 11 above)
Comment (znep) [inline attributes/default.rb]: Are you sure it isn't set by the postgres cookbook before we can get in to change the version?

That is why we had to duplicate some version related config for userdbs.

Comment (JoeNunnelley) [inline attributes/default.rb]: it works for both 9.3 and 9.4 with the current version of the file.

**Pull request merged**